explore/src/main/scala/explore/Routing.scala

```diff
@@ -86,77 +90,96 @@ object Routing {
     RouterWithPropsConfigDsl[Page, ReuseView[RootModel]].buildConfig { dsl =>
       import dsl._
 
+      // We can't use gid.regexPattern because capture groups don't work here.
+      def gidRegEx[Id](implicit gid: Gid[Id]): String =
+        s"${gid.tag.value.toString}-[1-9a-f][0-9a-f]*"
+
       def id[Id](implicit gid: Gid[Id]): StaticDsl.RouteB[Id] =
-        // We can't use gid.regexPattern because capture groups don't work here.
-        string(s"${gid.tag.value.toString}-[1-9a-f][0-9a-f]*").pmapL(gid.fromString)
+        string(gidRegEx).pmapL(gid.fromString)
```

Comment:
oh what is this? is `string` defined in `scalajs-react`?

Reply:
Yeah. `string` is part of the scalajs-react routing DSL. It essentially extracts a bit of the route based on the regex. You can modify further with the `pMap*` methods.